### ISSUE TITLE
URDF importer: importing duplicate assets

### ISSUE THREAD

**Issue by pawelbudziszewski** [o3de/o3de-extras#289]
This issue concerns URDF importer in version #234 

When importing URDF, the importer creates a folder with imported assets (meshes). Such folders should be unique for different URDFs, but re-importing the same URDF uses the same assets folder. What's more, its content is removed during the import procedure. This may cause a lot of problems, including crash, in some cases when assets from this folder are being used by any entities on the scene.

**Steps to reproduce the crash:**
1. Import panda from [this urdf](https://github.com/o3de/o3de-physics-test-scene/tree/main/Assets/URDF/panda_description/panda_description) - this will create folder with assets and a prefab on the scene.
2. Modify the `panda.urdf` file by changing `hand.dae` to `finger.dae` in line 416
3. Import panda again

Crash in the moment of creating assets

**Comment by michalpelka** [comment]
Crash with this particular case occurs deep in the engine:
```
AZStd::vector<AZ::EventHandler<AZ::Data::Asset<AZ::RPI::ModelAsset> const>*, AZStd::allocator>::push_back(AZ::EventHandler<AZ::Data::Asset<AZ::RPI::ModelAsset> const>*&&) (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/std/containers/vector.h:0)
AZ::Event<AZ::Data::Asset<AZ::RPI::ModelAsset> const>::Connect(AZ::EventHandler<AZ::Data::Asset<AZ::RPI::ModelAsset> const>&) const (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/EBus/Event.inl:0)
AZ::EventHandler<AZ::Data::Asset<AZ::RPI::ModelAsset> const>::Connect(AZ::Event<AZ::Data::Asset<AZ::RPI::ModelAsset> const>&) (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/EBus/Event.inl:127)
AZ::Render::ModelReloader::ConnectOnReloadedEventHandler(AZ::EventHandler<AZ::Data::Asset<AZ::RPI::ModelAsset> const>&) (/home/michal/github/o3de/Gems/Atom/Feature/Common/Code/Source/Mesh/ModelReloader.cpp:49)
AZ::Render::ModelReloaderSystem::ReloadModel(AZ::Data::Asset<AZ::RPI::ModelAsset>, AZ::EventHandler<AZ::Data::Asset<AZ::RPI::ModelAsset> const>&) (/home/michal/github/o3de/Gems/Atom/Feature/Common/Code/Source/Mesh/ModelReloaderSystem.cpp:26)
AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6::operator()() (/home/michal/github/o3de/Gems/Atom/Feature/Common/Code/Source/Mesh/MeshFeatureProcessor.cpp:915)
decltype(InvokeTraits::forward<AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6>(fp)()) AZStd::Internal::INVOKE<AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6>(AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6&&) (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/std/typetraits/invoke_traits.h:208)
AZStd::invoke_result<AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6>::type AZStd::invoke<AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6>(AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6&&) (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/std/function/invoke.h:54)
void AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::SystemTickEvents, AZ::SystemTickEvents>, AZ::BusInternal::EBusImplTraits<AZ::SystemTickEvents, AZ::SystemTickEvents>>::QueueFunction<AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6>(AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6&&)::'lambda'()::operator()() (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/EBus/BusImpl.h:875)
decltype(InvokeTraits::forward<AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6>(fp)()) AZStd::Internal::INVOKE<void AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::SystemTickEvents, AZ::SystemTickEvents>, AZ::BusInternal::EBusImplTraits<AZ::SystemTickEvents, AZ::SystemTickEvents>>::QueueFunction<AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6>(AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6&&)::'lambda'()&>(AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6&&) (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/std/typetraits/invoke_traits.h:208)
AZStd::invoke_result<AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6>::type AZStd::invoke<void AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::SystemTickEvents, AZ::SystemTickEvents>, AZ::BusInternal::EBusImplTraits<AZ::SystemTickEvents, AZ::SystemTickEvents>>::QueueFunction<AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6>(AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6&&)::'lambda'()&>(AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6&&) (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/std/function/invoke.h:54)
void AZStd::Internal::function_util::invoke_void_return_wrapper<void>::call<void AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::SystemTickEvents, AZ::SystemTickEvents>, AZ::BusInternal::EBusImplTraits<AZ::SystemTickEvents, AZ::SystemTickEvents>>::QueueFunction<AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6>(AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6&&)::'lambda'()&>(void AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::SystemTickEvents, AZ::SystemTickEvents>, AZ::BusInternal::EBusImplTraits<AZ::SystemTickEvents, AZ::SystemTickEvents>>::QueueFunction<AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6>(AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6&&)::'lambda'()&) (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/std/function/function_template.h:41)
void AZStd::Internal::function_util::get_invoker<void (), AZStd::Internal::function_util::function_obj_tag, AZStd::stateless_allocator>::call<void AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::SystemTickEvents, AZ::SystemTickEvents>, AZ::BusInternal::EBusImplTraits<AZ::SystemTickEvents, AZ::SystemTickEvents>>::QueueFunction<AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6>(AZ::Render::ModelDataInstance::MeshLoader::OnCatalogAssetAdded(AZ::Data::AssetId const&)::$_6&&)::'lambda'()>(AZStd::Internal::function_util::function_buffer&) (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/std/function/function_template.h:147)
AZStd::function_intermediate<void>::operator()() const (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/std/function/function_template.h:604)
AZStd::function<void ()>::operator()() const (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/std/function/function_template.h:684)
AZ::EBusQueuePolicy<true, AZ::EBus<AZ::SystemTickEvents, AZ::SystemTickEvents>, AZStd::mutex>::Execute() (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/EBus/Policies.h:266)
AZ::BusInternal::EBusBroadcastQueue<AZ::EBus<AZ::SystemTickEvents, AZ::SystemTickEvents>, AZ::BusInternal::EBusImplTraits<AZ::SystemTickEvents, AZ::SystemTickEvents>>::ExecuteQueuedEvents() (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/EBus/BusImpl.h:366)
AZ::ComponentApplication::TickSystem() (/home/michal/github/o3de/Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp:1511)
CCryEditApp::IdleProcessing(bool) (/home/michal/github/o3de/Code/Editor/CryEdit.cpp:2354)
CCryEditApp::OnIdle(int) (/home/michal/github/o3de/Code/Editor/CryEdit.cpp:2269)
Editor::EditorQtApplication::maybeProcessIdle() (/home/michal/github/o3de/Code/Editor/Core/QtEditorApplication.cpp:317)
QtPrivate::FunctorCall<QtPrivate::IndexesList<>, QtPrivate::List<>, void, void (Editor::EditorQtApplication::*)()>::call(void (Editor::EditorQtApplication::*)(), Editor::EditorQtApplication*, void**) (/home/michal/.o3de/3rdParty/packages/qt-5.15.2-rev8-linux/qt/include/QtCore/qobjectdefs_impl.h:152)
void QtPrivate::FunctionPointer<void (Editor::EditorQtApplication::*)()>::call<QtPrivate::List<>, void>(void (Editor::EditorQtApplication::*)(), Editor::EditorQtApplication*, void**) (/home/michal/.o3de/3rdParty/packages/qt-5.15.2-rev8-linux/qt/include/QtCore/qobjectdefs_impl.h:185)
QtPrivate::QSlotObject<void (Editor::EditorQtApplication::*)(), QtPrivate::List<>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*) (/home/michal/.o3de/3rdParty/packages/qt-5.15.2-rev8-linux/qt/include/QtCore/qobjectdefs_impl.h:418)
QtPrivate::QSlotObjectBase::call(QObject*, void**) (@QSingleShotTimer::timerEvent(QTimerEvent*):37)
QSingleShotTimer::timerEvent(QTimerEvent*) (@QSingleShotTimer::timerEvent(QTimerEvent*):33)
```
Everything works fine when the User will remove the imported prefab before reimporting. Otherwise, it will try to update already existing source assets decreasing the stability of the whole engine. 
If the option "import meshes during import" is unticked the crash does not occur.

**Comment by michalpelka** [comment]
Fixed  with #297
